Orbit.h

```diff
@@ -1,122 +1,104 @@
-//----------------------------------------
-// Orbit Profiler
-// Copyright Pierric Gimmig 2013-2017
-//----------------------------------------
-#pragma once
-
-//-----------------------------------------------------------------------------
-// Orbit API
-//
-// Simply include this header in your project, there is no lib and no cpp file.
-//
-// MANUAL INSTRUMENTATION:
-// The main feature of Orbit is its ability to dynamically instrument functions
-// without having to recompile or even relaunch your application.  However,
-// if you still want to manually instrument your code, you can.
-//
-// Use ORBIT_SCOPE or ORBIT_START/ORBIT_STOP *macros* only.
-// DO NOT use OrbitScope(...)/OrbitStart(...)/OrbitStop() directly.
-// NOTE: You need to use string literals.  Dynamic strings are not supported
-// yet.
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_H_
+#define ORBIT_H_
+
+#include <stdint.h>
+
+// Orbit API (header-only)
 //
-// DLL LOADING:
-// If you want to control the loading of Orbit64.dll, i.e. you don't want to
-// have to manually inject it into your application, use the OrbitAPI class.
-// Calling OrbitAPI::Connect(...) will load the dll and try to connect to the
-// Orbit instance specified in parameters.  Note that Orbit64.dll needs to be
-// next to your exectutable.  You can also change the code below
-// (OrbitAPI::Init()).
-//-----------------------------------------------------------------------------
-
-#define ORBIT_SCOPE(name) OrbitScope ORBIT_UNIQUE(ORB)(ORBIT_LITERAL(name))
-#define ORBIT_START(name) OrbitStart(ORBIT_LITERAL(name))
-#define ORBIT_STOP OrbitStop()
-#define ORBIT_SEND(ptr, num) OrbitSendData(ptr, num)
-
-//-----------------------------------------------------------------------------
-struct OrbitAPI {
-  static inline bool Connect(const char* a_Host, int a_Port = 1789);
-  static inline bool IsConnected();
-
- private:
-  static inline void Init();
-
-  typedef void (*InitRemote)(char*);
-  typedef bool (*GetBool)();
-
-  static inline HINSTANCE& GetHandle();
-  static inline InitRemote& GetInitRemote();
-  static inline GetBool& GetIsConnected();
-};
+// While the main feature of Orbit is its ability to dynamically instrument
+// functions, manual instrumentation is still possible using the macros below.
+// These macros call empty functions that Orbit dynamically instruments.
 
-//-----------------------------------------------------------------------------
-#define ORBIT_NOOP       \
-  static volatile int x; \
-  x;
-#define ORBIT_LITERAL(x) ("" x)
+// To disable manual instrumentation macros, set ORBIT_API_ENABLED to 0.
```

Comment:
You mean the macro should be re-defined?

Reply:
I mean replace 1 by 0 in the line below. I changed the comment to "...define ORBIT_API_ENABLED as 0".  The plan is to change the value directly in the header, maybe we should remove it from the header and let the user define it to enable manual instrumentation.

Reply:
Yeah that makes sense.  I prefer having it off by default actually so that the user explicitly has to turn on the feature that incurs a small overhead.  There will be a few iterations on this file anyway, I'll remove the define shortly.